MohammadBallal/task_manager.py

```diff
@@ -0,0 +1,138 @@
+from task import Task, UrgentTask
+from utils import validate_date, get_valid_input, valid_priority ,validate_status
+
+
+regularTasks= {}
+urgentTasks= {}
+
+def add_task():
+    #using get_valid_input for all user inputs to validate it
+    title= get_valid_input("What is the task title? : ")
+    description= get_valid_input("Just a little description for the task here please : ")
+    due_date= get_valid_input("Enter task due date (DD/MM/YYYY)")
+    #validate date format
+    validate_date(due_date)
+    #check if the task is urgent or not to know where to store it 
+    isUrgent= get_valid_input("Is this task urgent (please write : yes || no) :  ").lower() #To reduce user input errors
+    if isUrgent == "yes" :
+        priority= get_valid_input("Enter task priority ( High || Medium || Low ) : ").lower()
+        #validate priority
+        valid_priority(priority)
+        task= UrgentTask(title,description,due_date,priority)
+        urgentTasks[task.task_id] = task
+    else :
+        task= Task(title,description,due_date)
+        regularTasks[task.task_id] = task
+    
+    print("TASK ADDED SUCCESSFULLY")
+
+
+def view_tasks():
+    #Check if there any reguler tasks stored 
+    if regularTasks:
+        #Display every single task by display method from Task class
+        for task in regularTasks.values():
+            print(f"{task.display()}\n")
+            
+    else:
+        print("There is no reguler tasks here")
+    #Check if there any urgent task stored 
+    if urgentTasks:
+        for task in urgentTasks.values():
+            print(f"{task.display()}\n")
+    else:
+        print("There is no urgent tasks here")
+
+
+def update_task():
```

Comment:
what if the task is urgent and i need to update the priority?

Reply:
I work with urgent tasks as if there was no need to change the priority since it’s just a priority and is set at the beginning to focus on it.
I’ve modified the code to allow for priority changes.